GitUI/CommandsDialogs/BrowseDialog/FormCommitCount.cs

```diff
@@ -28,69 +35,113 @@ private void cbIncludeSubmodules_CheckedChanged(object sender, EventArgs e)
 
         private void FetchData()
         {
-            Loading.Visible = true;
+            ThreadHelper.ThrowIfNotOnUIThread();
+
+            _backgroundLoaderTokenSource.Cancel();
+            _backgroundLoaderTokenSource = new CancellationTokenSource();
+            var token = _backgroundLoaderTokenSource.Token;
+
+            ThreadHelper.JoinableTaskFactory.RunAsync(
+                async () =>
+                {
+                    Loading.Visible = true;
+
+                    CommitCount.Text = "";
+
+                    var includeSubmodules = cbIncludeSubmodules.Checked;
+
+                    await TaskScheduler.Default;
+
+                    var text = GenerateText(Module, includeSubmodules, token);
+
+                    await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync(token);
+
+                    if (!token.IsCancellationRequested)
+                    {
+                        CommitCount.Text = text;
+                        Loading.Visible = false;
+                    }
+                })
+                .FileAndForget();
+        }
+
+        private static string GenerateText(GitModule module, bool includeSubmodules, CancellationToken token)
+        {
+            var text = new StringBuilder();
+            var submodulesByName = new Dictionary<string, HashSet<string>>();
+            var (countByName, _) = CommitCounter.GroupAllCommitsByContributor(module);
 
-            CommitCount.Text = "";
-            var dict = new Dictionary<string, HashSet<string>>();
-            var (items, _) = CommitCounter.GroupAllCommitsByContributor(Module);
-            if (cbIncludeSubmodules.Checked)
+            if (includeSubmodules)
             {
-                var submodules = Module.GetSubmodulesLocalPaths();
+                token.ThrowIfCancellationRequested();
+
```

Comment:
:bulb: Would be good to remove this blank line to indicate the placement of the cancellation check is associated with the line that follows it. (Same applies to other cases.)